smart-contracts/test/Lock/upgrades/v13.js

```diff
@@ -101,7 +106,8 @@ describe('PublicLock upgrade v12 > v13', () => {
 
       // deploy new implementation
       lock = await upgrades.upgradeProxy(lock.address, PublicLockLatest, {
-        unsafeSkipStorageCheck: true, // UNSECURE - but we need the flag as we are resizing the `__gap`
```

Comment:
here is one main reason we didn't catch that error... 😓

Reply:
Actually I think we need to change `scripts/lock/testUpgrade.js` to make sure it runs with any/all versions

Reply:
In the test/Lock/upgrades we have all versions increment so that should cover it, but yes we should merge all that into a single one. The only thing is that we may have specific migrate logic so that's why they are separated